File: Latihan3/Latihan3_CPP/Pc.cpp
#include <iostream>
#include <string>
#include "Processor.cpp"
#include "Ram.cpp"
#include "Disk.cpp"

using namespace std;

class Pc
{
private:
	/* data */
	// class attribute
	Processor processor;
	Ram ram;
	Disk disk;
	int totalPrice;

public:
	Pc(/* args */){};
	// constructors
	Pc(Processor processor, Ram ram, Disk disk)
	{
		this->disk = disk;
		this->ram = ram;
		this->processor = processor;
		this->totalPrice = disk.getPrice() + ram.getPrice() + disk.getPrice();
	};

	// getter and setter
	void setProcessor(Processor processor)
	{
		this->processor = processor;
	};

	Processor getProcessor()
	{
		return this->processor;
	};

	void setRam(Ram ram)
	{
		this->ram = ram;
	};

	Ram getRam()
	{
		return this->ram;
	};

	void setDisk(Disk disk)
	{
		this->disk = disk;
	};

	Disk getDisk()
	{
		return this->disk;
	};

	void setTotalPrice(int processorPrice, int ramPrice, int diskPrice)
	{
		this->totalPrice = processorPrice + ramPrice + diskPrice;
	}

	int getTotalPrice()
	{
		return this->totalPrice;
	}

	// show class method
	void ShowClass()
	{
		cout << "	PC SPECS" << endl;
		cout << "PC Price	: " << this->totalPrice << endl;
		cout << "=========================\n";
		cout << "	PROCESSOR\n";
		cout << "=========================\n";
		cout << "Processor Name  : " << this->processor.getName() << endl;
		cout << "Processor Price : " << this->processor.getPrice() << endl;
		cout << "=========================\n";
		cout << "	DISK\n";
		cout << "=========================\n";
		cout << "Disk Type	: " << this->disk.getType() << endl;
		cout << "Disk Capacity	: " << this->disk.getCapacity() << endl;
		cout << "Disk Price	: " << this->disk.getPrice() << endl;
		cout << "=========================\n";
		cout << "	RAM\n";
		cout << "=========================\n";
		cout << "Ram Capacity	: " << this->ram.getCapacity() << endl;
		cout << "Ram Price	: " << this->ram.getPrice() << endl;
	}

	// destructors
	~Pc(){};
};

File: Latihan3/Latihan3_Java/Pc.java
public class Pc {

	// class attribute
	Processor processor;
	Disk disk;
	Ram ram;
	int totalPrice;

	// constructors
	Pc() {
	}

	Pc(Processor processor, Disk disk, Ram ram) {
		this.processor = processor;
		this.disk = disk;
		this.ram = ram;
		this.totalPrice = processor.getPrice() + disk.getPrice() + ram.getPrice();
	}

	// getter and setter
	public void setProcessor(Processor processor) {
		this.processor = processor;
	}

	public Processor getProcessor() {
		return processor;
	}

	public void setDisk(Disk disk) {
		this.disk = disk;
	}

	public Disk getDisk() {
		return disk;
	}

	public void setRam(Ram ram) {
		this.ram = ram;
	}

	public Ram getRam() {
		return ram;
	}

	public void setTotalPrice(int processorPrice, int diskPrice, int ramPrice) {
		this.totalPrice = processorPrice + diskPrice + ramPrice;
	}

	public int getTotalPrice() {
		return totalPrice;
	}

	// show class method
	public void ShowClass() {
		System.out.println("	PC SPECS");
		System.out.println("PC Price	: " + this.totalPrice);
		System.out.println("=========================");
		System.out.println("	PROCESSOR");
		System.out.println("=========================");
		System.out.println("Processor	: " + this.processor.getName());
		System.out.println("Processor Price : " + this.processor.getPrice());
		System.out.println("=========================");
		System.out.println("	DISK");
		System.out.println("=========================");
		System.out.println("Disk Type 	: " + this.disk.getType());
		System.out.println("Disk Capacity   : " + this.disk.getCapacity());
		System.out.println("Disk Price 	: " + this.disk.getPrice());
		System.out.println("=========================");
		System.out.println("	RAM");
		System.out.println("=========================");
		System.out.println("RAM Capacity 	: " + this.ram.getCapacity());
		System.out.println("RAM Price	: " + this.ram.getPrice());
	}
}
